### ISSUE TITLE
Form saves to database

### ISSUE THREAD

**Issue by cjhskippy** [18F/cms-carts-seds#475]
As a state user, I want my responses to be saved as I progress through the form so that I don't have to enter the entire report in one marathon session.

As a state user, I only want authorized users to be able to save changes to my state's report.

Current design choice for this interaction may need discussion: will we auto save or have a save button?  (see discussion in #256 )

A/C:
- [x] Only authorized users are able to save form responses.
- [x] Users are notified if there is an error saving the form
- [x] Users are provided visible updates when the form is saved.
- Replaced with #783 ~Answers provided by user are saved to persistent storage.  Upon logging out and visiting the same page a second time, the saved answers are loaded.~

Depends on user authentication #392

**Comment by mgwalker** [comment]
The remaining acceptance criteria are blocked by getting auth tied in on the backend side. The only frontend change that should be necessary will be having it post data to the real URL (instead of the dummy one created for this PR).

**Comment by timoballard** [comment]
#705 is ready for review/merge. Adds saving functionality to the API without authentication or authorization checks, as those are being implemented more broadly as separate items (see #736 and #737)

**Comment by timoballard** [comment]
This was mistakenly autoclosed by #705 - reopening

**Comment by jgillis1** [comment]
@timoballard Could you confirm the saving is occurring? I was on mdctcartsdev.cms.gov and attempted to change data, navigated to another site, came back and my change wasn't shown. After clicking around on the page a few times, this error message popped up:
![image.png](https://images.zenhubusercontent.com/5f11af98ebec4c4565352e89/da251b9f-e81e-49cd-8a4f-b3e69af2350f)

**Comment by mgwalker** [comment]
We're removing the acceptance criteria related to the form status, right? In which case, since saving is working, it looks like all that's left is the "authorized users" piece. Is that right @timoballard?

@StephanieCoppel If ☝️  is right, I think moving this to MGR review would be appropriate, and we can have a separate issue for making sure that only authorized users can actually do the save. I suspect a whole lot of auth stuff will turn on all at once. It's a work tracking question, so however you want to proceed – but the form is saving to the database now. 😄

**Comment by timoballard** [comment]
That’s correct. The auth portion is progressing more generally in #736. The plan is to integrate that with @tadhg-ohiggins user permissions work (#737), at which point a lot of this should snap into place
